test/Tests/src/Semantic/Hashcode_Spec.enso

```diff
@@ -0,0 +1,34 @@
+from Standard.Test import Test
+
+type My_Type
+    Value foo bar baz
+
+spec =
+    Test.specify "Hashcode and equality contract" <|
```

Comment:
Please avoid testing `Any.hash_code` in Enso tests. Only then you'll be able to remove `hash_code` and make it really internal.